files/lib/system/endpoint/controller/faq/questions/search/GetSearch.class.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace wcf\system\endpoint\controller\faq\questions\search;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Override;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use wcf\data\faq\Question;
+use wcf\data\faq\QuestionList;
+use wcf\http\Helper;
+use wcf\system\endpoint\GetRequest;
+use wcf\system\endpoint\IController;
+use wcf\system\exception\UserInputException;
+use wcf\system\WCF;
+
+#[GetRequest('/faq/questions/search')]
+final class GetSearch implements IController
+{
+    #[Override]
+    public function __invoke(ServerRequestInterface $request, array $variables): ResponseInterface
+    {
+        $parameters = Helper::mapApiParameters($request, GetSearchParameters::class);
+        if (\mb_strlen($parameters->query) < 3) {
+            throw new UserInputException('query', 'tooShort');
+        }
+
+        $questionIDs = $this->getQuestionsIDs($parameters->query);
+
+        return new JsonResponse([
+            'template' => WCF::getTPL()->fetch('faqQuestionSearchResult', 'wcf', [
+                'questions' => $this->getQuestions($questionIDs),
+            ], true),
```

Comment:
You don’t need to enable the sandbox here because this runs in its entirely own context. The sandbox is only required for things like boxes or BBCodes that can appear in arbitrary positions and thus run the risk of overwriting existing variables.